### PR TITLE
Fixed threading Error Occurring at the End of the Train

### PR DESCRIPTION
**Pull request by DrewNF**
I use tf.RandomShuffleQueue instead of tf.FIFOQueue deleted the use of
the coordinator and all the try except blocks, due that were useless if
everything works; I put all threads as Daemon so their lifecycle
depends from the one of the main thread.
Finally I use tf.train.start_queue_runners(sess=sess) to be sure that
all of them was working properly.

### CODE REVIEW

**Comment by DrewNF** [comment]
I also delete the lines:
 d = gen.next()
 sess.run(enqueue_op[phase], feed_dict=make_feed(d))
inside the loop:
with tf.Session(config=config) as sess:
        threads = []
        for phase in ['train', 'test']:
Seems to me useless, because when the thread starts make the same sess.run,
Let me know if work for you.

Bests

**Comment by Russell91** [comment]
I see, looks like the daemon is the correct way to force close the enque threads on program exits. But is there a reason to switch from FIFO to RandomShuffle? The data has already been shuffle in utils and all things being equal I think having more determinism is simpler.

**Comment by DrewNF** [comment]
If i'm not wrong RandomShuffle has embedded a FIFOQueue and many people on the net prefer to use it instead of FIFOQueue.
I think both are fine and equal, but maybe the RandomShuffleQueue give you the possibility to lightweight more the code, due to the embedded shuffuling.

**Comment by Russell91** [comment]
Okay, I merged this without the RandomShuffleQueue changes here: https://github.com/Russell91/TensorBox/commit/aab8e42c12d20c5e9acaa933535ea1696d66c608
